cfn-resources/cloud-outage-simulation/cmd/resource/resource.go

```diff
@@ -0,0 +1,324 @@
+	return handler.ProgressEvent{}, errors.New("not implemented: Update")
+}
+
+// List handles the List event from the Cloudformation service.
+func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	return handler.ProgressEvent{}, errors.New("not implemented: List")
+}
+
+func newOutageFilters(currentModel *Model) []atlasSDK.AtlasClusterOutageSimulationOutageFilter {
+	outageFilters := make([]atlasSDK.AtlasClusterOutageSimulationOutageFilter, 0)
+
+	for ind := range currentModel.OutageFilters {
+		mMatcher := atlasSDK.AtlasClusterOutageSimulationOutageFilter{
+			CloudProvider: currentModel.OutageFilters[ind].CloudProvider,
+			RegionName:    currentModel.OutageFilters[ind].Region,
+			Type:          currentModel.OutageFilters[ind].Type,
+		}
+		outageFilters = append(outageFilters, mMatcher)
+	}
+	return outageFilters
+}
+
+// function to track snapshot creation status
+func validateProgress(client *util.MongoDBClient, currentModel *Model, targetState string) (handler.ProgressEvent, error) {
+	projectID := *currentModel.ProjectId
+	clusterName := *currentModel.ClusterName
+	isReady, state, err := isCompleted(client, projectID, clusterName, targetState)
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	if !isReady {
+		p := handler.NewProgressEvent()
+		p.ResourceModel = currentModel
+		p.OperationStatus = handler.InProgress
+		p.CallbackDelaySeconds = 65
+		p.Message = "Pending"
+		p.CallbackContext = map[string]interface{}{
+			"status":       state,
+			"cluster_name": *currentModel.ClusterName,
+			"project_id":   *currentModel.ProjectId,
+		}
+		return p, nil
+	}
+
+	p := handler.NewProgressEvent()
+	if targetState != Complete {
+		p.ResourceModel = currentModel
+	} else {
+		p.ResourceModel = nil
+	}
+	p.OperationStatus = handler.Success
+	p.Message = "Complete"
+	return p, nil
+}
+
+// function to check if resource action is completed
+func isCompleted(client *util.MongoDBClient, projectID, clusterName, targetState string) (isExist bool, status string, err error) {
+	outageSimulation, resp, err := client.AtlasV2.ClusterOutageSimulationApi.GetOutageSimulation(context.Background(), projectID, clusterName).Execute()
+	if err != nil {
+		if resp.StatusCode == 404 {
+			return true, "COMPLETED", nil
+		}
+		return false, "", err
+	}
+	if *outageSimulation.State != "" {
+		log.Printf("[DEBUG] status for MongoDB cluster outage simulation: %s: %s", clusterName, *outageSimulation.State)
+	}
+	return *outageSimulation.State == targetState, *outageSimulation.State, nil
+}
+
+// function to check if resource action is active
+func isActive(client *util.MongoDBClient, projectID, clusterName, targetState string) (isExist bool, status string, err error) {
+	outageSimulation, resp, err := client.AtlasV2.ClusterOutageSimulationApi.GetOutageSimulation(context.Background(), projectID, clusterName).Execute()
+	if err != nil {
+		if resp.StatusCode == 404 {
+			return false, "COMPLETED", nil
```

Comment:
we can get all of this status in a enum, lets not use the Utils enum, lets create e new one, inside this resource

Reply:
fixed